### PR TITLE
add db_connection_provider option for DBIC

### PR DESCRIPTION
**Pull request by joshrabinowitz**

### CODE REVIEW

**Comment by joshrabinowitz** [comment]
thoughts, @bigpresh ? I'm migrating to DBIC and didn't want to maintain two database connections

File: lib/Dancer/Plugin/SimpleCRUD.pm [hunk 1]
```diff
@@ -710,12 +721,23 @@ register_hook(qw(
 ));
 register_plugin;
 
+
+sub _database {
+    my %args = @_;
+    if ($args{db_connection_provider} && $args{db_connection_provider} eq "DBIC") {
+        require Dancer::Plugin::DBIC;
+        return Dancer::Plugin::DBIC::schema($args{db_connection_name})->storage->dbh;
+    } else {
```
Comment (bigpresh) [inline lib/Dancer/Plugin/SimpleCRUD.pm]: I think I'd prefer it to squawk loudly if `db_connection_provider` is set, but to a value we don't recognise - e.g. someone passed `DBIX` or something, and wonders why it was silently ignored and D::P::D used anyway.

File: t/lib/TestAppDBIC.pm [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    qq/insert into users values (4, 'bodger', '$password')/,
+    qq/insert into users values (5, 'mousey', '$password')/,
+    qq/insert into users values (6, 'mystery2', '$password')/,
+    qq/insert into users values (7, 'mystery1', '$password')/,
+);
+
+schema->storage->dbh->do($_) for @sql;
+
+# now set up our simple_crud interface, and use DBIC
+simple_crud( prefix => '/users',  record_title=>'A', db_table => 'users', editable => 0, db_connection_provider => "DBIC" );
```
Comment (bigpresh) [inline t/lib/TestAppDBIC.pm]: The repetition in this file irks me a bit - it'd be nice if a lot of that could be moved to a superclass with `t::lib::TestAppDatabase` and `t::lib::TestAppDBIC` being subclasses of it (or, it being a simple app that is agnostic to the provider used, and the test script that uses it sets the appropriate config before loading it, so it's the exact same code, and we just test that it behaves the same for both providers).

**Comment by bigpresh** [comment]
In general, yes!  Good stuff!  A couple of minor niggles regarding validation and copy & pasting test code, but nothing major.

I'd *prefer* it to avoid making DBIC a hard dependency if possible, though - that said, I don't think DBIC has too heinous a dependency chain.

**Comment by joshrabinowitz** [comment]
@bigpresh all good input, thank you

File: lib/Dancer/Plugin/SimpleCRUD.pm (hunk 1 above)
Comment (joshrabinowitz) [reply lib/Dancer/Plugin/SimpleCRUD.pm]: agreed, fixed

**Comment by joshrabinowitz** [comment]
I've made D:P:DBIC a soft dependency. Test code is definitely a little less clear as a result.

Let me know if I should simplify the code and make D:P:DBIC a hard dependency.

**Comment by joshrabinowitz** [comment]
ok, I think this is pretty close to merge-able.

**Comment by joshrabinowitz** [comment]
I think this is good to go! 
Would like to get this merged and then add tests for D:P:Auth::Extensible via both D:P:Database and D:P:PDBIC

**Comment by bigpresh** [comment]
Excellent stuff!

**Pull request merged**